subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ExecuteWorkBuildOperationFiringStep.java

```diff
@@ -39,29 +40,31 @@ public ExecuteWorkBuildOperationFiringStep(BuildOperationExecutor buildOperation
 
     @Override
     public R execute(UnitOfWork work, C context) {
-        return operation(operationContext -> {
-                R result = delegate.execute(work, context);
-                ExecuteWorkBuildOperationType.Result operationResult = new ExecuteWorkResult(result.getExecution());
-                operationContext.setResult(operationResult);
-                return result;
-            },
-            BuildOperationDescriptor
-                .displayName("Execute unit of work")
-                .details(new ExecuteWorkDetails(work)));
+        Optional<String> buildOperationWorkType = work.getBuildOperationWorkType();
+        return buildOperationWorkType.map(workType -> operation(operationContext -> {
```

Comment:
🧐 Any reason to extract this to a separate local variable?

Reply:
Nope, not really. Apart from making the one expression a little bit shorter.